src/qt/test/apptests.cpp

```diff
@@ -63,6 +64,7 @@ void AppTests::appTests()
 #endif
 
     BasicTestingSetup test{CBaseChainParams::REGTEST}; // Create a temp data directory to backup the gui settings to
```

Comment:
Duplicate testing setups are illegal anyway. Have you seen https://github.com/bitcoin/bitcoin/pull/18923/commits/fa5f7fa24cc5ec34f3ba9616e6941123f751d8ed which is a massively smaller diff, potentially fixing the issue you ran into.

Reply:
#18923